includes/queue/class-wc-queue.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * WC Queue
+ *
+ * @version 3.5.0
+ * @package WooCommerce/Interface
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * WC Queue
+ *
+ * Singleton for managing the WC queue instance.
+ *
+ * @version 3.5.0
+ */
+class WC_Queue {
+
+	/**
+	 * The single instance of the queue.
+	 *
+	 * @var WC_Queue_Interface|null
+	 */
+	protected static $instance = null;
+
+	/**
+	 * The default queue class to initialize
+	 *
+	 * @var string
+	 */
+	protected static $default_cass = 'WC_Action_Queue';
+
+	/**
+	 * @return WC_Queue_Interface
+	 */
+	final static function instance() {
+
+		if ( is_null( self::$instance ) ) {
+			$class = self::get_class();
+			self::$instance = new $class();
+			self::$instance = self::validate_instance( self::$instance );
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Get class to instantiate
+	 *
+	 * And make sure 3rd party code has the chance to attach a custom queue class.
+ 	 *
+	 * @return string
+	 */
+	protected static get_class() {
```

Comment:
Missing function declaration

Reply:
🤦‍♂️ by "I haven't tested this changeset much yet", what I meant was "at all yet". :)

I've fixed that up and rebased in the fix. :)